### PR TITLE
Convert hui-input-select-entity-row to TypeScript/LitElement

### PR DESCRIPTION
**Pull request by iantrich**

### CODE REVIEW

File: src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+          slot="dropdown-content"
+          selected="${stateObj.attributes.options.indexOf(stateObj.state)}"
+        >
+          ${
+            repeat(
+              stateObj.attributes.options,
+              (option) =>
+                html`
+                  <paper-item
+                    @click="${() => this._selectedChanged(option as string)}"
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts]: Don't do this. Instead mimic how the old component behaved: observing `selected-item-label` (as seen by it using `{…}` which indicates 2 way databinding). You can do this by listening to the `selected-item-label-changed` event. Don't attach individual listeners to each item.

File: src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    `;
+  }
+
+  private _selectedChanged(option: string): void {
+    // Selected Option will transition to '' before transitioning to new value
+    const stateObj = this.hass!.states[this._config!.entity];
+    if (option === "" || option === stateObj.state) {
+      return;
+    }
+    this.hass!.callService("input_select", "select_option", {
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts]: I think that we should extract this into a method into the `data` folder like we did for shopping list. New file `input_select.ts` with

```
export const selectOption = (hass, entity, option) => hass.callService(…)
```

Comment (balloob) [reply src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts]: Eventually we will have all data mutation in a single place, making it easy to manage.

File: src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts (hunk 1 above)
Comment (iantrich) [reply src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts]: Hopefully I didn't botch that too much with the commit I just pushed (it works) :) I was pretty unsure of what I was doing with this listener.

**Review by balloob — APPROVED** [review]

**Pull request merged**